GSA_oM/Elements/RigidConstraint.cs

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using System;
+using System.ComponentModel;
+using System.Collections.Generic;
+using BH.oM.Quantities.Attributes;
+using BH.oM.Structure.Elements;
+using BH.oM.Adapters.GSA.SpacerProperties;
+using BH.oM.Analytical.Elements;
+
+namespace BH.oM.Adapters.GSA.Elements
+{
+    [Description("A rigid constraint defining linkage between nodes.")]
+    public class RigidConstraint : BHoMObject
```

Comment:
Not to keen of the addition of this class is it fully overlaps, both in structure as well as intent, with the [RigidLink](https://github.com/BHoM/BHoM/blob/develop/Structure_oM/Elements/RigidLink.cs).

Another solution to this is to have RigidConstraints as well as Link elements both being read when asking for RigidLinks.

Can add a Fragment or similar in GSA_oM highlighting the type they actually refer to, and that can also help when pushing, dispatching out either to create a link element or RigidConstraint in GSA.

Probably warrants a call through how to handle this.

Reply:
Based on the code, there are no new elements introduced that match the `RigidLink` and the fragment approach appears to have been adopted.

Reply:
This appears to have been removed when looking at the code but still comes up in the files changes:
![image](https://github.com/BHoM/GSA_Toolkit/assets/30078587/b2c38d45-e533-403c-98b3-9fd51d7aed7d)